Src/AutoFixture/DomainNameGenerator.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Creates new <see cref="DomainName"/> instances.
+    /// </summary>
+    public class DomainNameGenerator : ISpecimenBuilder
+    {
+        private readonly string[] fictitiousDomains =
+        {
+            "example.com",
+            "example.net",
+            "example.org"
+        };
+
+        private readonly Random random = new Random();
+
+        /// <summary>
+        /// Creates a new specimen based on a request.
+        /// </summary>
+        /// <param name="request">The request that describes what to create</param>
+        /// <param name="context">A context that can be used to create other specimens.</param>
+        /// <returns>
+        /// The requested specimen if possible; otherwise a <see cref="NoSpecimen"/> instance.
+        /// </returns>
+        public object Create(object request, ISpecimenContext context)
+        {
+            if (request == null || !typeof(DomainName).Equals(request))
+            {
+                return new NoSpecimen();
```

Comment:
Well, now you mention it, I had already seen it, when looking around in the code elsewhere. However, as you mention, it's easy to do wrong, and APIs, of course, should make it easy to do the right thing, and difficult to do the wrong thing ;)

Looking at the references of `Request`, it's indeed not being used (at least, not in the `AutoFixture.sln` solution, I'll do a more thorough check, if I pick it up :smiley:).

![image](https://cloud.githubusercontent.com/assets/460904/11381561/fbe0e8c8-92fc-11e5-86e2-578111b52a9a.png)